crates/rust-analyzer/src/semantic_tokens.rs

```diff
@@ -12,6 +12,7 @@ pub(crate) const TYPE_ALIAS: SemanticTokenType = SemanticTokenType::new("typeAli
 pub(crate) const UNION: SemanticTokenType = SemanticTokenType::new("union");
 pub(crate) const UNRESOLVED_REFERENCE: SemanticTokenType =
     SemanticTokenType::new("unresolvedReference");
+pub(crate) const FORMAT_SPECIFIER: SemanticTokenType = SemanticTokenType::new("formatSpecifier");
```

Comment:
Could you add it to `SUPPORTED_MODIFIERS` bellow as well? Also, do you have any ideas how to make "forgot to add to the list of tokens" error impossible? 

I did this myself a coulpe of times, and that's a pretty bug to have, as the syntax highlgithing will die horribly in runtime. 

At the moment, I'm leaning towards hacking the "one definition" with a macro, but perhaps there's a simpler solution?...

Reply:
bors d+

Reply:
s/MODIFIERS/the other thing which is for types/

Reply:
One could add a test case that checks for every syntax range that the `HighlightTag` is contained in `SUPPORTED_TYPES`. That would be rather simple to implement and would spot those kind of mistakes (assuming that the HighlighTag is actually used in the syntax highlighting tests).